syntax = "proto3";

package aws_message;

message stdev_block {
    optional float average = 1;  // average over the measurement period
    optional float minus_std_dev = 2;  // average value minus standard deviation of observations over measurement period
    optional float plus_std_dev = 3;  // average value plus standard deviation of observations over measurement period
}

message min_max_block {
    optional float average = 1;  // average over the measurement period
    optional float max = 2;  // maximum value in the measurement period
    optional float min = 3;  // minimum value in the measurement period
}

message pressure_block {
    optional float qfe = 1; // units=hectopascals
    optional float qff = 2; // units = hectopascals
    optional float qnh = 3; // units = hectopascals
}

message engineering_stats { //Values useful for troubleshooting or other support function. Not Meteorologial observations
    optional float battery_voltage = 1; //Output voltage on internal UPS battery, units=V
    optional float internal_temp = 2; // Temperature inside AWS cabinet, units=degrees C
}

message aws_record {
    optional string station_wsi = 1; //WIGOS Station Identifier
    optional int64 obs_dt_utc = 2; // date and time of observation in UTC units: milliseconds since Epoch
    optional min_max_block wind_speed = 4; // Wind Speed at 10m, Units=knots
    optional stdev_block wind_direction = 5; // Wind Direction at 10m, Units = degrees from magnetic north.
    optional min_max_block air_temperature=6; //air_temperature at 2m, Units=degrees Celsius
    optional min_max_block dew_point=7; //dew point temperature at 2m, Units=degrees Celsius
    optional min_max_block rel_humidity=8; //relative humidity at 2m, Units=%
    repeated float clouds = 9; // height observations from ceilometer, Units=feet
    optional float visibility = 10; //MOR, Units=m
    optional float rainfall = 11; //accumulation (resets at 9999.9), Units=mm
    optional pressure_block pressure = 12; // barometric pressure and aviation related derived values, Units=hectopascals
    optional engineering_stats engineering = 13; // data to facilitate engineering support of equipment
}

message aws_message_new{
    repeated aws_record records = 1;
}
